ipaserver/plugins/dogtag.py

```diff
@@ -1575,49 +1460,41 @@ def get_certificate(self, serial_number):
         """
         logger.debug('%s.get_certificate()', type(self).__name__)
 
-        # Convert serial number to integral type from string to properly handle
-        # radix issues. Note: the int object constructor will properly handle large
-        # magnitude integral values by returning a Python long type when necessary.
-        serial_number = int(serial_number, 0)
-
         # Call CMS
-        http_status, _http_headers, http_body = (
-            self._sslget('/ca/agent/ca/displayBySerial',
-                         self.env.ca_agent_port,
-                         serialNumber=str(serial_number),
-                         xml='true')
+        path = 'certs/{}'.format(serial_number)
+        _http_status, _http_headers, http_body = self._ssldo(
+            'GET', path, use_session=False,
+            headers={
+                'Accept': 'application/json',
+            },
         )
 
-
-        # Parse and handle errors
-        if http_status != 200:
-            self.raise_certificate_operation_error('get_certificate',
-                                                   detail=http_status)
-
-        parse_result = self.get_parse_result_xml(http_body, parse_display_cert_xml)
-        request_status = parse_result['request_status']
-        if request_status != CMS_STATUS_SUCCESS:
-            self.raise_certificate_operation_error('get_certificate',
-                                                   cms_request_status_to_string(request_status),
-                                                   parse_result.get('error_string'))
+        try:
+            resp = json.loads(ipautil.decode_json(http_body))
+        except ValueError:
+            raise errors.RemoteRetrieveError(
+                reason=_("Response from CA was not valid JSON"))
 
         # Return command result
         cmd_result = {}
 
-        if 'certificate' in parse_result:
-            cmd_result['certificate'] = parse_result['certificate']
+        if 'Encoded' in resp:
+            s = resp['Encoded']
+            match = x509.PEM_CERT_REGEX.search(s.encode('utf-8'))
+            if match:
+                s = match.group(2).decode('utf-8')
+            cmd_result['certificate'] = s.strip()
```

Comment:
I don't quite understand why the regexp is necessary here. Could you please add a comment to educate your fellow developers what is going on here?

Reply:
sure, it's just stripping the PEM headers; the old method to do that was removed.  I'll add a comment.